app.models.js

```diff
@@ -6,4 +6,18 @@ function fetchTopics() {
   });
 }
 
-module.exports = fetchTopics;
+function fetchArticleById(aritcleId) {
+  return db
+    .query("SELECT * FROM articles WHERE article_id = $1;", [aritcleId])
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({ status: 404, msg: "Not Found" });
+      }
+      return rows[0];
+    })
+    .catch((err) => {
```

Comment:
You don't need this catch block here. The one in your controller will catch the error.